sqlmesh/core/plan/builder.py

```diff
@@ -137,6 +137,7 @@ def __init__(
             environment_catalog_mapping or {},
             name=self._context_diff.environment,
             suffix_target=environment_suffix_target,
+            normalize_name=self._context_diff.is_new_environment,
```

Comment:
is this for backwards compatibility? how does this work?

Reply:
Correct, new environments should be normalized but old ones shouldn't.

Reply:
why aren't old environments normalized? is this permanent?

Reply:
I think this may be incorrect after all, thanks for double checking it. I'll revisit.

Reply:
Addressed a bit differently, see https://github.com/TobikoData/sqlmesh/pull/2722/commits/9ebcc31f5f2b75addda9c12af838a29cea0e2bd1.

TL;DR added new field (`normalize_environment_name: bool`) in `ContextDiff` which is now set as:

```
is_new_environment or bool(env and env.normalize_name)
```

So, if an environment is new, we use the new logic. Otherwise, we check if the fetched environment's `normalize_name` column is `TRUE`.